src/weakptr.c

```diff
@@ -163,20 +163,22 @@ Int GrowWPObj (
 ** WP object 
 */
 
-Obj FuncWeakPointerObj( Obj self, Obj list ) { 
+#ifdef BOEHM_GC
+/* We need to make sure that <list> stays live until
+ * after REGISTER_WP(); on architectures that pass
+ * arguments in registers (x86_64, SPARC, etc), the
+ * argument register may be reused. In conjunction with
+ * loop unrolling, the reference to <list> may then be
+ * destroyed before REGISTER_WP() is called.
+ */
+Obj FuncWeakPointerObj( Obj self, volatile Obj list )
```

Comment:
Good question...

That said, I am not sure the previous code did what we want -- in particular, I didn't see anything in the C, see e.g. https://stackoverflow.com/questions/27816535

Maybe we need to rethink this completely...

Anyway, a "safer" fix than this PR might be to change the code to actually use `list2` -- in this function that should be easy.